src/common/Singleton.h

```diff
@@ -26,20 +28,24 @@ class Singleton {
   Singleton() noexcept = default;
 
   Singleton(const Singleton&) = delete;
-
   Singleton& operator=(const Singleton&) = delete;
 
+  Singleton(Singleton&&) = delete;
+  Singleton& operator=(Singleton&&) = delete;
+
   virtual ~Singleton() = default;  // to silence base class Singleton<T> has a
                                    // non-virtual destructor [-Weffc++]
 
  public:
-  static T& GetInstance() noexcept(std::is_nothrow_constructible<T>::value) {
-    // Guaranteed to be destroyed.
-    // Instantiated on first use.
-    // Thread safe in C++11
-    static T instance;
-
-    return instance;
+  static T& GetInstance(const std::function<std::shared_ptr<T>()>& _allocator =
```

Comment:
Any need to remove this? https://stackoverflow.com/questions/11711920/how-to-implement-multithread-safe-singleton-in-c11-without-using-mutex

This is how it is in C++11

Reply:
there was no reason to remove this, have added it back